presto-spi/src/main/java/io/prestosql/spi/Page.java

```diff
@@ -34,6 +34,15 @@
 {
     public static final int INSTANCE_SIZE = ClassLayout.parseClass(Page.class).instanceSize();
 
+    /**
+     * Visible to give trusted classes like {@link PageBuilder} access to a constructor that doesn't
+     * defensively copy the blocks
+     */
+    static Page ofBlocksWithoutCopy(int positionCount, Block[] blocks)
```

Comment:
Can you merge this commit into the last one, rename this method to `wrapBlocksWithoutCopy`, and then use this method in page instead of the constructor. I think this will make the `Page` code more readable because it will clearly state the intent to avoid the extra copy.

Reply:
Done.